en/02_Developer_Guides/03_Forms/07_Using_GridField_With_Arbitrary_Data.md

```diff
@@ -0,0 +1,437 @@
+
+```php
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+$gridField->getConfig()->removeComponentsByType(GridFieldFilterHeader::class);
+```
+
+### Filtering data {#arraydata-filter}
+
+If you want to be able to filter your `GridField`, you will need to tell the `GridField` which fields to filter against and how to do so. The `GridFieldFilterHeader` uses a [`SearchContext`](api:SilverStripe\ORM\Search\SearchContext) implementation to do most of the heavy lifting.
+
+Specifically the [`BasicSearchContext`](api:SilverStripe\ORM\Search\BasicSearchContext) is designed to be used for data that isn't represented by `DataObject` records.
+
+[hint]
+The `BasicSearchContext` respects all [customisations for the general search field](/developer_guides/model/scaffolding/#general-search-field). You can therefore set the relevant configuration properties on the `ArrayData` class via YAML configuration and it will be respected.
+
+The only such customisation it doesn't respect is changing the name of the field based on the class you're using to represent your data (i.e. setting the `general_search_field_name` configuration property on your data class). Instead, you can customise the name of the field using the [`BasicSearchContext.general_search_field_name`](api:api:SilverStripe\ORM\Search\BasicSearchContext->general_search_field_name) configuration property.
+[/hint]
+
+```php
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\Forms\HiddenField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\ORM\Search\BasicSearchContext;
+
+// Instantiate a BasicSearchContext and tell it which fields to search against
+$searchContext = BasicSearchContext::create(null);
+$searchFields = [
+    HiddenField::create(BasicSearchContext::config()->get('general_search_field_name')),
+    TextField::create('FeldName', 'Search Field Label'),
+];
+// Pass the BasicSearchContext into the GridFieldFilterHeader component
+$searchContext->setFields(FieldList::create($searchFields));
+$gridField->getConfig()->getComponentByType(GridFieldFilterHeader::class)->setSearchContext($searchContext);
+```
+
+### Exporting data {#arraydata-export}
+
+If you want to export or print your data, you don't have to do anything special - just make sure to include the `GridFieldExportButton` and `GridFieldPrintButton` components.
+
+[info]
+These will both use the field list you [passed into `GridFieldDataColumns`](#arraydata-display-as-rows) to know which fields they should use - though you can explicitly call [`GridFieldExportButton::setExportColumns()`](api:SilverStripe\Forms\GridField\GridFieldExportButton::setExportColumns()) and [`GridFieldPrintButton::setPrintColumns()`](api:SilverStripe\Forms\GridField\GridFieldPrintButton::setPrintColumns()) if you want to export/print different columns than those displayed in the grid view.
+[/info]
```

Comment:
I'm not sure why this needs to be in an info block. The info in this paragraph seems like a natural continuation from the previous paragraph.


```suggestion
`GridFieldExportButton` and `GridFieldPrintButton` default to outputting the field list provided to  [`GridFieldDataColumns`](#arraydata-display-as-rows). You can explicitly call [`GridFieldExportButton::setExportColumns()`](api:SilverStripe\Forms\GridField\GridFieldExportButton::setExportColumns()) and [`GridFieldPrintButton::setPrintColumns()`](api:SilverStripe\Forms\GridField\GridFieldPrintButton::setPrintColumns()) if you want to export/print different columns.
```

Reply:
Done but moved to be _below_ the example. It was in an `info` box to avoid separating the "make sure to include the things" part from the example.